### ISSUE TITLE
Content coded on vscode doesn't show saved on a jupyter notebook opened separately.

### ISSUE THREAD

**Issue by narbhar** [microsoft/vscode-jupyter#5174]
Issue Type: <b>Bug</b>

Open the folder workspace where the ipynb file is.
Open it to edit in Vscode. 
After a while open a seperate jupyter session on browser and open the same jupyter notebook being edited in vscode.
Notice it'd be behind several save versions and prepare for disappointment because it now overrides vscode save and all the code is now lost.
I think this is an issue in the autosave discrepancy between vscode and jupyter notebook on a browser.

Extension version: 2021.3.619093157
VS Code version: Code 1.54.2 (fd6f3bce6709b121a895d042d343d71f317d74e7, 2021-03-11T00:56:19.848Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.73GB (3.89GB free)|
|Process Argv|--crash-reporter-id b7305d1a-d3b9-4eed-a023-f034ca814d30|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspyt653:30270858
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry914:30256638
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
dockersubset:30270708
pythontbcf:30265426
vspre833:30267464
vsdfh931:30275552

```

</details>

**Comment by IanMatthewHuff** [comment]
@narbhar Notebook files should be portable between VS Code and browser based Jupyter sessions. However there isn't any type of specific "syncing" mechanism between VS Code and browser sessions. You would be at the mercy of when you are saving to make sure that you don't lose changes. If you have a repro where VS Code is failing to save when you believe that it should have I can help look into that. But if the issues is just the timing of saving then I would recommend not working with a notebook file in both VS Code and a browser at the exact same time.

**Comment by narbhar** [comment]
Oh that's a bit unfortunate to know there isn't any syncing mechanism. I think the least VS Code could do is to implement a checkpoint and autosave mechanism as the browser based jupyter session so that in such cases, it'd be easier to restore any saved content in case it gets overridden for some reason.

**Comment by DonJayamanne** [comment]
@narbhar Out of curiosity why are you using both VS Code & Jupyter to work on the same notebook.
What's missing or not working in VS Code that makes you want to use Jupyter at the same time?
Thanks

**Comment by narbhar** [comment]
@DonJayamanne I was investigating a separate issue where my jupyter session would time out with the following error:

`ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
Exception in callback BaseSelectorEventLoop._read_from_self()
handle: <Handle BaseSelectorEventLoop._read_from_self()>
Traceback (most recent call last):
  File "C:\Users\bnarapar\Anaconda3\envs\env8520gpu\lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\bnarapar\Anaconda3\envs\env8520gpu\lib\asyncio\selector_events.py", line 119, in _read_from_self
    data = self._ssock.recv(4096)`

This caused loss of kernel and leading to any data loss on executed cells. This affected VS Code too where it would say "unexpected host extension termination". I tried searching for it online but no one ever seemed to have faced this timeout issue.

Hence to investigate it further and check if the issue was with my particular python environment, I opened the same file running in VS Code in another environment as a browser session which is where I found what happens when you do that the hard way. The issue was ultimately present across all my python environments BTW.

I reset my whole python installation via anaconda and switched to miniconda now. Looks to be holding out fine for now.
